File: app/src/main/java/com/checkin/app/checkin/Shop/Private/Invoice/RestaurantSessionModel.java
package com.checkin.app.checkin.Shop.Private.Invoice;

import com.checkin.app.checkin.Misc.BriefModel;
import com.checkin.app.checkin.Utility.Utils;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;
import java.util.Date;

@JsonIgnoreProperties(ignoreUnknown = true)
@JsonInclude(JsonInclude.Include.NON_NULL)
public class RestaurantSessionModel implements Serializable {
    @JsonProperty("pk")
    private long pk;

    @JsonProperty("hash_id")
    private String hashId;

    @JsonProperty("count_orders")
    private int countOrders;

    @JsonProperty("count_customers")
    private int countCustomers;

    @JsonProperty("total")
    private double total;

    @JsonProperty("host")
    private BriefModel host;

    @JsonProperty("table")
    private String table;

    @JsonProperty("checked_in")
    private Date checkedIn;

    @JsonProperty("checked_out")
    private Date checkedOut;

    public RestaurantSessionModel() {
    }

    public long getPk() {
        return pk;
    }

    public String getHashId() {
        return hashId;
    }

    public int getCountOrders() {
        return countOrders;
    }

    public int getCountCustomers() {
        return countCustomers;
    }

    public double getTotal() {
        return total;
    }

    public BriefModel getHost() {
        return host;
    }

    public String getTable() {
        return table;
    }

    public Date getCheckedIn() {
        return checkedIn;
    }

    public Date getCheckedOut() {
        return checkedOut;
    }

    public String getFormattedDate() {
        return Utils.formatCompleteDate(checkedOut);
    }

    public String formatTotal() {
        return String.valueOf(total);
    }
}

File: app/src/main/java/com/checkin/app/checkin/Shop/Private/Invoice/RestaurantSessionModel.kt
package com.checkin.app.checkin.Shop.Private.Invoice

import com.checkin.app.checkin.Shop.ShopModel.PAYMENT_MODE
import com.checkin.app.checkin.user.models.UserBriefModel
import com.checkin.app.checkin.utility.Utils
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import java.io.Serializable
import java.util.*

data class RestaurantSessionModel(
        val pk: Long,
        @JsonProperty("hash_id") val hashId: String,
        @JsonProperty("count_orders") val countOrders: Int = 0,
        @JsonProperty("count_customers") val countCustomers: Int = 0,
        val total: Double,
        val host: UserBriefModel?,
        val table: String?,
        @JsonProperty("checked_in") val checkedIn: Date?,
        @JsonProperty("checked_out") val checkedOut: Date,
        @JsonProperty("payment_mode")
        @JsonDeserialize(using = PAYMENT_MODE.PaymentModeDeserializer::class)
        val paymentMode: PAYMENT_MODE?,
        @JsonProperty("checkedin_time") val workingCheckedinTime: Date,
        @JsonProperty("is_scheduled") val isScheduled: Boolean
) : Serializable {

    val formatHashId = "${if (isScheduled) "Booked" else "Session"} ID: #$hashId"

    val formattedDate: String = Utils.formatCompleteDate(if (isScheduled) workingCheckedinTime else checkedOut)

    val tableInfo: String = table ?: "Booked"

    fun formatTotal(): String = total.toString()
}